Controllers/FileController.cs

```diff
@@ -23,27 +80,91 @@ public async Task<ActionResult> Post()
                 Directory.CreateDirectory(uploadPath);
             }
 
-            var path = Path.Combine(uploadPath, file.FileName);
+            foreach (var file in files)
+            {
+                var path = Path.Combine(uploadPath, file.FileName);
+
+                using (var stream = new FileStream(path, FileMode.Create))
+                {
+                    await file.CopyToAsync(stream);
+                }
+            }
+            string uploaderName = form["uploader"];
+            string desserts = form["dessert"];  // Assuming this is a string like "dessert1,dessert2,dessert3"
+            string weather = form["weather"];
+            string country = form["country"];
+
+            // Split the desserts string into an array of individual desserts
+            string[] dessertArray = desserts.Split(',');
 
-            using (var stream = new FileStream(path, FileMode.Create))
+            // Loop through the array and add each dessert individually
+            foreach (string dessert in dessertArray)
             {
-                await file.CopyToAsync(stream);
+                // Increment a vote counter for the type of dessert
+                if (this.dessertVotes.ContainsKey(dessert))
+                {
+                    this.dessertVotes[dessert] += 1;
+                }
+                else
+                {
+                    this.dessertVotes[dessert] = 1;
+                }
             }
+            // Update the weather
+            currentWeather = weather;
+
+            // If author is new add them, their country of origin and update latest submission timestamp
+            var author = authorsList.Find(a => a.Name == uploaderName);
+            if (author == null)
+            {
+                authorsList.Add(new Author 
+                { 
+                    Name = CultureInfo.CurrentCulture.TextInfo.ToTitleCase(uploaderName),
+                    CountryOfOrigin = CultureInfo.CurrentCulture.TextInfo.ToTitleCase(country),
+                    LastSubmissionTimestamp = DateTime.Now
+                }); 
+            }
+            else
+            {
+                author.LastSubmissionTimestamp = DateTime.Now;
+            }
+
+            // Update allData before serializing it
+            allData = new
+            {
+                DessertVotes = this.dessertVotes,
+                CurrentWeather = currentWeather,
+                AuthorsList = authorsList
+            };
+
+            // Serialize the allData object to JSON
+            var json = JsonSerializer.Serialize(allData);
+            Console.WriteLine(json);
+            // Write the JSON to the file
+            await System.IO.File.WriteAllTextAsync(FilePath, json);
```

Comment:
Since we are using the file as a makeshift db, you will at least want to wrap all reads/writes in a lock to handle concurrent requests.